### PR TITLE
chore: refactor docker integration code

### PR DESCRIPTION
**Pull request by aslushnikov**
This patch splits out `dockerApi` namespace that is solely
responsible for docker API manipulation.

### CODE REVIEW

File: packages/playwright-test/src/docker/dockerApi.ts [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+// See https://docs.docker.com/engine/api/v1.41/
+
+const DOCKER_API_VERSION = '1.41';
+
+export interface DockerImage {
+  imageId: string;
+  names: string[];
+}
+
+export interface DockerContainer {
```
Comment (mxschmitt) [inline packages/playwright-test/src/docker/dockerApi.ts:28]:
```suggestion
interface DockerContainer {
```

Comment (aslushnikov) [reply packages/playwright-test/src/docker/dockerApi.ts:28]: this is used in docker.ts

File: packages/playwright-test/src/docker/dockerApi.ts [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+export async function checkEngineRunning() {
+  try {
+    await callDockerAPI('get', '/info');
+    return true;
+  } catch (e) {
+    return false;
+  }
+}
+
+async function getJSON(url: string): Promise<any> {
```
[the remark this answers is not in the document]
Comment (aslushnikov) [reply packages/playwright-test/src/docker/dockerApi.ts:154]: callDockerApi is async!

File: packages/playwright-test/src/docker/dockerApi.ts (hunk 1 above)
Comment (mxschmitt) [reply packages/playwright-test/src/docker/dockerApi.ts:28]: I see no import or usage! but anyways a nit.

**Review by mxschmitt — APPROVED** [review]

**Review by dgozman — APPROVED** [review]

**Pull request merged**